crates/ef-testing/src/evm_sequencer/account/v1.rs

```diff
@@ -66,25 +38,70 @@ impl KakarotAccount {
         let mut storage = vec![
             starknet_storage!(ACCOUNT_EVM_ADDRESS, evm_address),
             starknet_storage!(ACCOUNT_IS_INITIALIZED, 1u8),
+            starknet_storage!(ACCOUNT_BYTECODE_LEN, code.len() as u32),
         ];
 
         // Write the nonce of the account is written to storage after each tx.
         storage.append(&mut vec![starknet_storage!(ACCOUNT_NONCE, nonce)]);
 
-        // Initialize the bytecode storage vars.
-        // Assumes that the bytecode is stored as a ByteArray type, following the Store<ByteArray> implementation of
-        // the cairo core library
-        let mut bytecode_storage = prepare_bytearray_storage(code);
+        // Initialize the bytecode storage var.
+        let mut bytecode_storage = pack_byte_array_to_starkfelt_array(code)
+            .enumerate()
+            .map(|(i, bytes)| (StorageKey::from(i as u32), bytes))
+            .collect();
         storage.append(&mut bytecode_storage);
 
+        // Initialize the code hash var
+        let account_is_empty =
+            code.is_empty() && nonce == Felt::from(0) && balance == U256::from(0);
+        let code_hash = if account_is_empty {
+            U256::from(0)
+        } else if code.is_empty() {
+            U256::from_be_slice(KECCAK_EMPTY.as_slice())
+        } else {
+            U256::from_be_slice(keccak256(code).as_slice())
+        };
+
+        let code_hash_values = split_u256(code_hash);
+        let code_hash_low_key = get_storage_var_address(ACCOUNT_CODE_HASH, &[]);
+        let code_hash_high_key = next_storage_key(&code_hash_low_key)?;
+        storage.extend([
+            (code_hash_low_key, Felt::from(code_hash_values[0])),
+            (code_hash_high_key, Felt::from(code_hash_values[1])),
+        ]);
+
+        // Initialize the bytecode jumpdests.
+        let bytecode = to_analysed(Bytecode::new_raw(code.clone()));
+        let valid_jumpdests: Vec<usize> = match bytecode {
+            Bytecode::LegacyAnalyzed(legacy_analyzed_bytecode) => legacy_analyzed_bytecode
+                .jump_table()
+                .0
+                .iter()
+                .enumerate()
+                .filter_map(|(index, bit)| bit.as_ref().then(|| index))
+                .collect(),
+            _ => unreachable!("Bytecode should be analysed"),
+        };
+
+        let jumdpests_storage_address = get_storage_var_address(ACCOUNT_VALID_JUMPDESTS, &[]);
+        let jumdpests_storage_address = *jumdpests_storage_address.0.key();
```

Comment:
```suggestion
        let jumdpests_storage_address = Felt::from(jumdpests_storage_address);
```
I think this work, imo clearer